fatrace.c

```diff
@@ -375,9 +375,12 @@ setup_fanotify (int fan_fd)
     while ((mount = getmntent (mounts)) != NULL) {
         /* Only consider mounts which have an actual device or bind mount
          * point. The others are stuff like proc, sysfs, binfmt_misc etc. which
-         * are virtual and do not actually cause disk access. */
-        if (mount->mnt_fsname == NULL || access (mount->mnt_fsname, F_OK) != 0 ||
-            mount->mnt_fsname[0] != '/') {
+         * are virtual and do not actually cause disk access. 
+         * zfs mount point don't start with a "/" so allow them anyway */
+        if (mount->mnt_fsname == NULL || 
+            (strcmp(mount->mnt_type, "zfs") != 0 && (access (mount->mnt_fsname, F_OK) != 0 || 
+                                                    mount->mnt_fsname[0] != '/'))
+            ) {
```

Comment:
I stared at this for a bit, and logically it's correct. However, I find it hard to read. It's also indented wrongly (off by one space).

I suggest to leave the original condition as-is, and instead check for zfs inside. Like this:

```C
if (original || condition) {
    /* zfs mount point don't start with a "/" so allow them anyway */
   if (strcmp(mount->mnt_type, "zfs") != 0) {
        debug();
        continue;
   }
}
```
WDYT?